src/Raven.Server/Documents/Handlers/TransactionsRecordingHandler.cs

```diff
@@ -0,0 +1,172 @@
+                var stopwatch = Stopwatch.StartNew();
+                stopwatch.Start();
+                foreach (var replayProgress in ReplayTxCommandHelper.Replay(Database, replayStream))
+                {
+                    commandsProgress = replayProgress.CommandsProgress;
+                    if (replayProgress.CommandsProgress > commandAmountForNextRespond)
+                    {
+                        commandAmountForNextRespond = replayProgress.CommandsProgress + commandAmountBetweenResponds;
+                        onProgress(new ReplayTxProgress
+                        {
+                            ProcessedCommand = replayProgress.CommandsProgress,
+                            PassedTime = stopwatch.Elapsed
+                        });
+                    }
+
+                    token.ThrowIfCancellationRequested();
+                }
+                stopwatch.Stop();
+
+                return new ReplayTxOperationResult
+                {
+                    ExecutedCommandsAmount = commandsProgress,
+                    PassedTime = stopwatch.Elapsed
+                };
+            }
+            catch (Exception e)
+            {
+                //Because the request is working while the file is uploading the server needs to ignore the rest of the stream
+                //and the client needs to stop sending it
+                HttpContext.Response.Headers["Connection"] = "close";
+                throw new InvalidOperationException("Failed to process replay transaction commands", e);
+            }
+        }
+
+        [RavenAction("/databases/*/admin/transactions/start-recording", "POST", AuthorizationStatus.ClusterAdmin)]
+        public async Task StartRecording()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var json = await context.ReadForMemoryAsync(RequestBodyStream(), null);
+                var parameters = JsonDeserializationServer.StartTransactionsRecordingOperationParameters(json);
+                if (parameters.File == null)
+                {
+                    ThrowRequiredPropertyNameInRequest(nameof(parameters.File));
+                }
+
+                var command = new TransactionsRecordingCommand(
+                        Database.TxMerger,
+                        TransactionsRecordingCommand.Instruction.Start,
+                        parameters.File
+                    );
+
+                await Database.TxMerger.Enqueue(command);
+            }
+        }
+
+        [RavenAction("/databases/*/admin/transactions/stop-recording", "POST", AuthorizationStatus.ClusterAdmin)]
+        public async Task StopRecording()
+        {
+            var command = new TransactionsRecordingCommand(
+                Database.TxMerger,
+                TransactionsRecordingCommand.Instruction.Stop
+            );
+
+            await Database.TxMerger.Enqueue(command);
+        }
```

Comment:
PUT/POST/etc need to return `NoContent` if they are not returning anything.

Use `NoContent` or `NoContentStatus` methods for this.

Reply:
done